libs/wingsdk/src/target-tf-aws/service.ts

```diff
@@ -0,0 +1,338 @@
+
+    let taskRole = new IamRole(this, "TaskRole", {
+      assumeRolePolicy: JSON.stringify({
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Effect: "Allow",
+            Principal: {
+              Service: "ecs-tasks.amazonaws.com",
+            },
+            Action: "sts:AssumeRole",
+          },
+        ],
+      }),
+      inlinePolicy: [{
+        name: `${this.assetName}-inline-policy`,
+        policy: Lazy.stringValue({
+          produce: () => {
+            this.policyStatments = this.policyStatments ?? [];
+            
+            if (this.policyStatments.length !== 0) {
+              return JSON.stringify({
+                Version: "2012-10-17",
+                Statement: this.policyStatments
+              });
+            }
+
+            return JSON.stringify({
+              Version: "2012-10-17",
+              Statement: [
+                {
+                  Effect: "Allow",
+                  Action: "none:null",
+                  Resource: "*"
+                }
+              ]
+            })
+          }
+        }),
+      }]
+    });
+
+    let logGroup = new CloudwatchLogGroup(this, "LogGroup", {
+      name: `/ecs/${this.assetName}`,
+      retentionInDays: 7,
+    });
+
+    let taskDef = new EcsTaskDefinition(this, "Task", {
+      family: this.assetName,
+      executionRoleArn: executionRole.arn,
+      taskRoleArn: taskRole.arn,
+      containerDefinitions: Lazy.anyValue({
+        produce: () => JSON.stringify([
+            {
+              name: this.assetName,
+              image: image.name,
+              environment: Object.entries(this._env).map(([key, value]) => {
+                return {
+                  name: key,
+                  value: value,
+                };
+              }
+              ),
+              logConfiguration: {
+                logDriver: "awslogs",
+                options: {
+                  "awslogs-group": `/ecs/${this.assetName}`,
+                  "awslogs-region": "us-east-1", // TODO: Can I ignore this?
+                  "awslogs-stream-prefix": logGroup.name,
+                }
+              }
+          }
+        ])
+      }) as any,
+      networkMode: "awsvpc",
+      requiresCompatibilities: ["FARGATE"],
+      memory: "512",
+      cpu: "256",
+    });
+
+    let subnetIds = app.subnets["private"].map((subnet) => subnet.id);
+
+    let sg = new SecurityGroup(this, "SecurityGroup", {
+      vpcId: app.vpc.id,
+      egress: [
+        {
+          fromPort: 0,
+          toPort: 0,
+          protocol: "-1",
+          cidrBlocks: ["0.0.0.0/0"],
+        }
+      ],
+      ingress: [
+        { // TODO: Support network config
+          fromPort: 0,
+          toPort: 0,
+          protocol: "-1",
+          cidrBlocks: ["0.0.0.0/0"]
+        }
+      ]
+    });
+
+    this.service = new EcsService(this, "Service", {
+      name: this.assetName,
+      cluster: app.ecsCluster.arn,
+      desiredCount: 1,
```

Comment:
This should be configurable through `cloud.Service`.

Reply:
Im working on a followup PR to adjust the way `cloud.Service` works theres a limitation in the current API, Ill plan to make these configurations there.

Basically when a return function is expected, we have an issue where a service that is long running I.E.

```js
let s = new cloud.Service(inflight () => {
  while true {
    let index = counter.inc();
    bucket.put("file-{index}.txt", "Hello, world!");
    util.sleep(10s);
  }

	// This will never get returned
  return inflight () => {
    for file in bucket.list() {
      bucket.delete(file);
    }
  };
});
```
The program loop never finishes, so calling stop on the service has no function to run when the service shuts down.